tasks/libs/common/git.py

```diff
@@ -1,18 +1,35 @@
 from __future__ import annotations
 
-import os.path
+import os
+import tempfile
+from contextlib import contextmanager
 from typing import TYPE_CHECKING
 
 from invoke.exceptions import Exit
 
-from tasks.libs.common.color import color_message
+from tasks.libs.common.color import Color, color_message
 from tasks.libs.common.constants import DEFAULT_BRANCH
 from tasks.libs.common.user_interactions import yes_no_question
 
 if TYPE_CHECKING:
     from collections.abc import Iterable
 
 
+@contextmanager
+def clone(ctx, repo, branch, options=""):
+    """
+    Context manager to clone a git repository and checkout a specific branch.
+    """
+    current_dir = os.getcwd()
+    try:
+        with tempfile.TemporaryDirectory() as clone_dir:
+            ctx.run(f"git clone -b {branch} {options} https://github.com/DataDog/{repo} {clone_dir}")
+            os.chdir(clone_dir)
+            yield
+    finally:
+        os.chdir(current_dir)
```

Comment:
nit - I think you could also `ctx.cd` and yield in the context manager, but this lgtm